pytest_django/runner.py

```diff
@@ -0,0 +1,64 @@
+import os
+from optparse import make_option
+
+import pytest
+
+
+class PyTestRunner(object):
+    """
+    Runs py.test to discover and run tests.
+    """
+    option_list = (
+        make_option('-t', '--top-level-directory',
+                    action='store', dest='top_level', default=None,
+                    help='Top level of project for unittest discovery.'),
+        make_option('-p', '--pattern', action='store', dest='pattern',
+                    default="test*.py",
+                    help='The test matching pattern. Defaults to test*.py.'),
+        make_option('-k', '--keepdb',
+                    action='store_true', dest='keepdb', default=False,
+                    help='Preserves the test DB between runs.'),
+    )
+
+    def __init__(self, pattern=None, top_level=None, verbosity=1,
+                 interactive=True, failfast=False, keepdb=False, **kwargs):
+        if pattern and pattern != 'test*.py':
+            # TODO: implement
+            raise NotImplementedError('Testing with a file pattern is not '
+                                      'implemented.')
+        if top_level is not None:
+            # TODO: implement
+            raise NotImplementedError('Specifying the top_level is not '
+                                      'implemented.')
+        self.verbosity = verbosity
+        self.interactive = interactive
+        self.failfast = failfast
+        self.keepdb = keepdb
+
+        # TODO: Make this an option
+        self.ds = os.environ['DJANGO_SETTINGS_MODULE']
+
+    def run_tests(self, test_labels, extra_tests=None, **kwargs):
```

Comment:
This is [the code that calls your runner from Django](https://github.com/django/django/blob/master/django/core/management/commands/test.py#L89-L90):

        test_runner = TestRunner(**options)
        failures = test_runner.run_tests(test_labels)

So while it may be wise to leave `**kwargs` for compatibility with future Django changes, there is no extra_tests passed in.